docs/book/how-to/setting-up-a-project-repository/shared-components-for-teams.md

```diff
@@ -0,0 +1,123 @@
+
+```bash
+zenml artifact-store flavor register <path.to.MyS3ArtifactStoreFlavor>
+```
+
+### Custom Steps
+
+Custom steps can be created and shared via a separate repository. Team members can reference these components as they would normally reference Python modules.
+
+### Custom Materializers
+
+Custom materializers are common components that teams often need to share. To implement and share a custom materializer:
+
+1. Create the materializer in a shared repository.
+2. Implement the custom materializer as described in the [ZenML documentation](https://docs.zenml.io/how-to/handle-data-artifacts/handle-custom-data-types).
+3. Team members can import and use the shared materializer in their projects.
+
+## How to Distribute Shared Components
+
+There are several methods to distribute and use shared components within a team:
+
+### Shared Private Wheels
+
+Using shared private wheels is an effective approach to sharing code within a team. This method packages Python code for internal distribution without making it publicly available.
+
+#### Benefits of Using Shared Private Wheels
+
+- Packaged format: Easy to install using pip
+- Version management: Simplifies managing different code versions
+- Dependency management: Automatically installs specified dependencies
+- Privacy: Can be hosted on internal PyPI servers
+- Smooth integration: Imported like any other Python package
+
+#### Setting Up Shared Private Wheels
+
+1. Create a private PyPI server or use a service like [AWS CodeArtifact](https://aws.amazon.com/codeartifact/).
+2. [Build your code](https://packaging.python.org/en/latest/tutorials/packaging-projects/) [into wheel format](https://opensource.com/article/23/1/packaging-python-modules-wheels).
+3. Upload the wheel to your private PyPI server.
+4. Configure pip to use the private PyPI server in addition to the public one.
+5. Install the private packages using pip, just like public packages.
+
+### Using Shared Libraries with `DockerSettings`
+
+When running pipelines with remote orchestrators, ZenML generates a Dockerfile at runtime. You can use the `DockerSettings` class to specify how to include your shared libraries in this Docker image.
+
+#### Installing Shared Libraries
+
+Here are some ways to include shared libraries using `DockerSettings`. Either specify a list of requirements:
+
+```python
+docker_settings = DockerSettings(requirements=["my-private-package==1.0.0"])
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+Or you can also use a requirements file:
+
+```python
+docker_settings = DockerSettings(requirements="/path/to/requirements.txt")
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+For information on using private PyPI repositories to share your code, see our [documentation on how to use a private PyPI repository](../customize-docker-builds/how-to-use-a-private-pypi-repository.md).
```

Comment:
Doesnt this need to be updated now based on the latest changes @strickvl

Reply:
Ok you want this section also to be about private repositories with logins. Got it.

Reply:
Done